Mathlib/Algebra/DualNumber.lean

```diff
@@ -95,40 +95,94 @@ theorem commute_eps_left [Semiring R] (x : DualNumber R) : Commute ε x := by
 /-- `ε` commutes with every element of the algebra. -/
 theorem commute_eps_right [Semiring R] (x : DualNumber R) : Commute x ε := (commute_eps_left x).symm
 
-/-- For two algebra morphisms out of `R[ε]` to agree, it suffices for them to agree on `ε`. -/
-@[ext]
-theorem algHom_ext {A} [CommSemiring R] [Semiring A] [Algebra R A] ⦃f g : R[ε] →ₐ[R] A⦄
-    (h : f ε = g ε) : f = g :=
-  algHom_ext' <| LinearMap.ext_ring <| h
+variable {A : Type*} [CommSemiring R] [Semiring A] [Semiring B] [Algebra R A] [Algebra R B]
+
+/-- For two `R`-algebra morphisms out of `A[ε]` to agree, it suffices for them to agree on the
+elements of `A` and the `A`-multiples of `ε`. -/
+@[ext 1100]
+nonrec theorem algHom_ext' ⦃f g : A[ε] →ₐ[R] B⦄
+    (hinl : f.comp (inlAlgHom _ _ _) = g.comp (inlAlgHom _ _ _))
+    (hinr : f.toLinearMap ∘ₗ (LinearMap.toSpanSingleton A A[ε] ε).restrictScalars R =
+        g.toLinearMap ∘ₗ (LinearMap.toSpanSingleton A A[ε] ε).restrictScalars R) :
+      f = g :=
```

Comment:
Sorry, I'm failing to understand what you generalised. What was assumed commutative before which is not anymore?

Reply:
The generalization is from `R[ε] →ₐ[R] B` to `A[ε] →ₐ[R] B`, where `A` is not commutative

Reply:
Added to the PR description

Reply:
Ah I see